tap_frontapp/streams.py

```diff
@@ -65,30 +66,55 @@ def select_fields(mdata, obj):
 @on_exception(constant, MetricsRateLimitException, max_tries=5, interval=60)
 @on_exception(expo, RateLimitException, max_tries=5)
 @sleep_and_retry
-@limits(calls=1, period=61) # 60 seconds needed to be padded by 1 second to work
-def get_metric(atx, metric, start_date, end_date):
-    LOGGER.info('Metrics query - metric: {} start_date: {} end_date: {} '.format(
-        metric,
-        start_date,
-        end_date))
-    return atx.client.get('/analytics', params={'start': start_date, \
-            'end': end_date, 'metrics[]':metric}, endpoint='analytics')
+@limits(calls=100, period=61) # 60 seconds needed to be padded by 1 second to work
+def get_metric(atx, metric, start_date, end_date, page_token = ''):
+    if metric == 'conversations':
+        LOGGER.info('Conversations query - metric: {} start_date: {} end_date: {} '.format(
+            metric,
+            start_date,
+            end_date))
+        return atx.client.get('/conversations', \
+                              params={'q[before]':end_date,'q[after]':start_date,'page_token':page_token,'limit':100}, \
+                              endpoint='conversations')
+    else:
+        LOGGER.info('Metrics query - metric: {} start_date: {} end_date: {} '.format(
+            metric,
+            start_date,
+            end_date))
+        return atx.client.get('/analytics', params={'start': start_date, \
+                'end': end_date, 'metrics[]':metric}, endpoint='analytics')
 
 def sync_metric(atx, metric, incremental_range, start_date, end_date):
     with singer.metrics.job_timer('daily_aggregated_metric'):
         start = time.monotonic()
         start_date_formatted = datetime.datetime.utcfromtimestamp(start_date).strftime('%Y-%m-%d')
         # we've really moved this functionality to the request in the http script
         #so we don't expect that this will actually have to run mult times
-        while True:
-            if (time.monotonic() - start) >= MAX_METRIC_JOB_TIME:
-                raise Exception('Metric job timeout ({} secs)'.format(
-                    MAX_METRIC_JOB_TIME))
-            data = get_metric(atx, metric, start_date, end_date)
-            if data != '':
-                break
-            else:
-                time.sleep(METRIC_JOB_POLL_SLEEP)
+
+        if metric == 'team_table':
+            while True:
+                if (time.monotonic() - start) >= MAX_METRIC_JOB_TIME:
+                    raise Exception('Metric job timeout ({} secs)'.format(
+                        MAX_METRIC_JOB_TIME))
+                data = get_metric(atx, metric, start_date, end_date)
+                if data != '':
+                    break
+                else:
+                    time.sleep(METRIC_JOB_POLL_SLEEP)
+        if metric == 'conversations':
```

Comment:
I'm seeing a pattern of "if team_table... if conversations..." It would be safer if the existing `sync_metric` (and related functions) was renamed to `sync_team_table` and the conversations metric just had its own sync functions. That way they can share state logic and the sync code can check at the spot linked below to dispatch between the metrics (and potential future metrics).
https://github.com/singer-io/tap-frontapp/blob/a7b43c9238f3637ff564956e2708d1c25db625ed/tap_frontapp/streams.py#L211

It also has the effect of reducing branching, which makes this code in its current state very hard to follow.